en/companion_computer/holybro_pixhawk_jetson_baseboard.md

```diff
@@ -0,0 +1,1143 @@
+:::
+
+With the help from [PX4 ROS 2 user guide](../ros2/user_guide.md#building-the-workspace) we could have a basebone to test communication between the agent (Jetson) and client (Pixhawk) using XRCE-DDS:
+
+```sh
+mkdir -p ~/ws_sensor_combined/src/
+cd ~/ws_sensor_combined/src/
+git clone https://github.com/PX4/px4_msgs.git
+git clone https://github.com/PX4/px4_ros_com.git
+cd ..
+source /opt/ros/humble/setup.bash
+colcon build
+```
+
+1- Setting the connection based on internal serial
+
+```sh
+param set MAV_1_CONFIG = 0 (Disabling current TELEM2 MAVLINK instance)
+param set UXRCE_DDS_CFG  102 (TELEM2)
+param set UXRCE_DDS_DOM_ID  0 (If you would like to change this you need to change your host domain ID as well. Check ROS2 Humble Domain ID)
+param set UXRCE_DDS_PTCFG 0
+param set UXRCE_DDS_SYNCC 0 (But if you are sure about your companion UTC time being synced and updated you could consider making this equal to 1 so that PX4 time is updated with your XRCE-DDS Agent time or if there are no external sources like hardware RTC or GPS device)
+param set UXRCE_DDS_SYNCT 1
+```
+
+::: info
+This might not be a recommended way to establish agent-client connection as it will occupy the only serial connection between pixhawk and Jetson.
+The reasons are:
+
+1. Serial connection has higher latency
+1. Reboots or system changes may cause failures to ethernet switch
+1. You could use the serial connection for lower level applications
+
+:::
+
+A Pixhawk reboot is needed after all these so that the changes are applied.
+After the reboot please check the client status by running the following the following inside MAVLINK shell:
+
+```sh
+uxrce_dds_client status
+```
+
+A healthy output should show:
+
+![uXRCE-DDS Client serial status](../../assets/companion_computer/holybro_pixhawk_jetson_baseboard/uxrce_dds_serial_agent_running.png)
+
+Also another way to check if the client has started running at the boot is to get `dmesg` output from MAVLINK shell.
+The below output as a part of `dmesg` log mentions the baudrate and the instance (`/dev/tty/S4` is equal to `TELEM2` for Holybro Pixhawk 6X and 6X Pro) the client is running on.
+
+```sh
+Starting UXRCE-DDS Client on /dev/ttyS4
+INFO  [uxrce_dds_client] init serial /dev/ttyS4 @ 921600 baud
+```
+
+::: info
+In case the client was not running you could run it manually:
+
+```sh
+uxrce_dds_client start -t serial -d /dev/ttyS4 -b 921600
+```
+
+:::
+
+After setting up the params and making sure the client is running we can start the agent on serial as below inside Jetson terminal:
+
+```sh
+sudo MicroXRCEAgent serial --dev /dev/ttyTHS1 -b 921600
+```
+
+2 - Setting the XRCE-DDS client connection to use Ethernet.
+Besides, we are choosing UXRCE_DDS_PRT to define a default udp port PX4 runs the client on Ethernet.
+
+```sh
+param set UXRCE_DDS_CFG 1000 (Ethernet)
+param set UXRCE_DDS_PRT 8888
+```
+
+Then we can run MicroXRCEAgent over Ethernet by:
+
+```sh
+MicroXRCEAgent udp4 -p 8888
+```
```

Comment:
What's the best way to start the client and agent on boot? Most users are happy to play with terminals when developing, but when the code is finished it will have to all run automagically.

Reply:
Ah, I see you cover that but just in the flow of text. We should make that a heading. I'll sort it.